lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -22,7 +23,11 @@ pub struct ProxySegment {
     pub write_segment: LockedSegment,
     pub wrapped_segment: LockedSegment,
     /// Points which should not longer used from wrapped_segment
+    /// May contain points which are not in wrapped_segment,
+    /// because the set is shared among all proxy segments
     deleted_points: LockedRmSet,
+    /// Number of points removed from this segment
+    deleted_points_count: AtomicUsize,
```

Comment:
If I understand this properly, this means we have the following invariant
`self.deleted_points_count <= self.deleted_points.len()`

Is it possible to have a sanity check at runtime and issue at least a log if there is an issue?